### ISSUE TITLE
Cannot cast java.lang.RuntimeException to java.sql.SQLException at java.base/java.lang.Class.cast(Class.java:3605)

### ISSUE THREAD

**Issue by gbhipolito** [aws/aws-advanced-jdbc-wrapper#1155]
### Describe the bug

Not sure where to ask this, I can't see an inquiry section.

I'm getting the error:

```
org.springframework.transaction.CannotCreateTransactionException: Could not open JPA EntityManager for transaction; nested exception is java.lang.ClassCastException: Cannot cast java.lang.RuntimeException to java.sql.SQLException
	at org.springframework.orm.jpa.JpaTransactionManager.doBegin(JpaTransactionManager.java:467)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.startTransaction(AbstractPlatformTransactionManager.java:400)
<... some more lines here ...>
```

and then

```
Caused by: java.lang.ClassCastException: Cannot cast java.lang.RuntimeException to java.sql.SQLException
	at java.base/java.lang.Class.cast(Class.java:3605)
	at software.amazon.jdbc.util.WrapperUtils.wrapExceptionIfNeeded(WrapperUtils.java:559)
	at software.amazon.jdbc.plugin.failover.FailoverConnectionPlugin.dealWithOriginalException(FailoverConnectionPlugin.java:536)
	at software.amazon.jdbc.plugin.failover.FailoverConnectionPlugin.execute(FailoverConnectionPlugin.java:200)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$execute$3(ConnectionPluginManager.java:337)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$makePluginChainFunc$2(ConnectionPluginManager.java:280)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$null$1(ConnectionPluginManager.java:280)
	at software.amazon.jdbc.plugin.AuroraConnectionTrackerPlugin.execute(AuroraConnectionTrackerPlugin.java:123)
<... some more lines here ...>
```

I can't get the root cause of the error or what the SQLException is about because it's being wrapped in another exception and then the wrapping caused an exception. And this happens intermittently so I don't know how to re-create it. Is there a way that I can get what the SQLException is about?

Thanks!

I'm using:
```xml
<groupId>org.springframework.boot</groupId>
<artifactId>spring-boot-starter-data-jpa</artifactId>

<groupId>org.springframework.boot</groupId>
<artifactId>spring-boot-starter</artifactId>

<groupId>org.postgresql</groupId>
<artifactId>postgresql</artifactId>
<version>42.6.1</version>

<groupId>software.amazon.jdbc</groupId>
<artifactId>aws-advanced-jdbc-wrapper</artifactId>
<version>2.2.2</version>
```

And the config properties:
```env
database.settings.driverClassName=software.amazon.jdbc.Driver
database.settings.jdbcUrl=jdbc:aws-wrapper:postgresql://${myhost}:${myport}/${mydatabase}?currentSchema=${myschema}
```

### Expected Behavior

I'm expecting it to not have this error, or at least a way to get what the SQLException is all about. Thanks!

### What plugins are used? What other connection properties were set?

aws-advanced-jdbc-wrapper 2.2.2, database.settings.driverClassName=software.amazon.jdbc.Driver. I've provided more details in the description.

### Current Behavior

Here are the stack traces:

```
org.springframework.transaction.CannotCreateTransactionException: Could not open JPA EntityManager for transaction; nested exception is java.lang.ClassCastException: Cannot cast java.lang.RuntimeException to java.sql.SQLException
	at org.springframework.orm.jpa.JpaTransactionManager.doBegin(JpaTransactionManager.java:467)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.startTransaction(AbstractPlatformTransactionManager.java:400)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.getTransaction(AbstractPlatformTransactionManager.java:373)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.createTransactionIfNecessary(TransactionAspectSupport.java:595)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:382)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:762)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:89)
	at my.app.yeah.config.helper.TargetDataSourceAspect.advice(TargetDataSourceAspect.java:24)
	at jdk.internal.reflect.GeneratedMethodAccessor124.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:762)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:762)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:707)
	at my.app.yeah.status.service.LightningStatusService$$EnhancerBySpringCGLIB$$7d71858d.getLatestStatus(<generated>)
	at my.app.yeah.event.driven.service.MyProcessingService.processAlertObj(MyProcessingService.java:186)
	at my.app.yeah.event.driven.service.MyProcessingService.processAlerts(MyProcessingService.java:131)
	at my.app.yeah.event.driven.service.MyProcessingService$WorkerThread.run(MyProcessingService.java:425)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

```
Caused by: java.lang.ClassCastException: Cannot cast java.lang.RuntimeException to java.sql.SQLException
	at java.base/java.lang.Class.cast(Class.java:3605)
	at software.amazon.jdbc.util.WrapperUtils.wrapExceptionIfNeeded(WrapperUtils.java:559)
	at software.amazon.jdbc.plugin.failover.FailoverConnectionPlugin.dealWithOriginalException(FailoverConnectionPlugin.java:536)
	at software.amazon.jdbc.plugin.failover.FailoverConnectionPlugin.execute(FailoverConnectionPlugin.java:200)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$execute$3(ConnectionPluginManager.java:337)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$makePluginChainFunc$2(ConnectionPluginManager.java:280)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$null$1(ConnectionPluginManager.java:280)
	at software.amazon.jdbc.plugin.AuroraConnectionTrackerPlugin.execute(AuroraConnectionTrackerPlugin.java:123)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$execute$3(ConnectionPluginManager.java:337)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$makePluginChainFunc$2(ConnectionPluginManager.java:280)
	at software.amazon.jdbc.ConnectionPluginManager.executeWithSubscribedPlugins(ConnectionPluginManager.java:258)
	at software.amazon.jdbc.ConnectionPluginManager.execute(ConnectionPluginManager.java:334)
	at software.amazon.jdbc.util.WrapperUtils.executeWithPlugins(WrapperUtils.java:219)
	at software.amazon.jdbc.util.WrapperUtils.runWithPlugins(WrapperUtils.java:161)
	at software.amazon.jdbc.wrapper.ConnectionWrapper.setAutoCommit(ConnectionWrapper.java:700)
	at com.zaxxer.hikari.pool.ProxyConnection.setAutoCommit(ProxyConnection.java:414)
	at com.zaxxer.hikari.pool.HikariProxyConnection.setAutoCommit(HikariProxyConnection.java)
	at org.hibernate.resource.jdbc.internal.AbstractLogicalConnectionImplementor.begin(AbstractLogicalConnectionImplementor.java:72)
	at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.begin(LogicalConnectionManagedImpl.java:285)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.begin(JdbcResourceLocalTransactionCoordinatorImpl.java:246)
	at org.hibernate.engine.transaction.internal.TransactionImpl.begin(TransactionImpl.java:83)
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.beginTransaction(HibernateJpaDialect.java:164)
	at org.springframework.orm.jpa.JpaTransactionManager.doBegin(JpaTransactionManager.java:421)
	... 28 common frames omitted
```

### Reproduction Steps

I'm not sure how to reproduce it as it's intermittent and I can't get the actual error message from SQLException

### Possible Solution

Could be connection issue? But how to know for sure?

### The AWS Advanced JDBC Driver version used

2.2.2

### JDK version used

dockerized: openjdk-11-slim

### Operating System and version

Debian GNU/Linux, "11 (bullseye)" via AWS

**Comment by aaron-congo** [comment]
Hi @gbhipolito, thanks for reaching out with this issue. I believe this problem was fixed in 2.2.3. However, 2.2.3 is fairly old, we  just recently released 2.5.0. Can you please try updating your version to 2.5.0 and see if this fixes the problem for you?

**Comment by gbhipolito** [comment]
Thanks for the quick response Aaron. I'll try this out.
